src/game/stage/results_init.c

```diff
@@ -0,0 +1,176 @@
+
+    if (NextTransitionFrame(ts) == SCREEN_TRANSITION_COMPLETE) {
+        TaskDestroy(gCurTask);
+
+        if (gGameMode == GAME_MODE_BOSS_TIME_ATTACK) {
+            CreateTimeAttackResults(gCourseTime);
+            return;
+        }
+
+        if (IS_FINAL_STAGE(gCurrentLevel)) {
+            return;
+        }
+
+        if ((gSelectedCharacter == CHARACTER_SONIC)
+            && gLoadedSaveGame->unlockedLevels[CHARACTER_SONIC] <= gCurrentLevel) {
+            switch (LEVEL_TO_ZONE(gCurrentLevel)) {
+                case ZONE_1: {
+                    CreateStageResultsCutscene(COURSE_END_UNLOCK_CREAM);
+                } break;
+
+                case ZONE_3: {
+                    CreateStageResultsCutscene(COURSE_END_UNLOCK_TAILS);
+                } break;
+
+                case ZONE_5: {
+                    CreateStageResultsCutscene(COURSE_END_UNLOCK_KNUCKLES);
+                } break;
+
+                default: {
+                    CreateStageResults(gCourseTime, gRingCount, gSpecialRingCount);
+                } break;
+            }
+
+            return;
+        } else {
+            CreateStageResults(gCourseTime, gRingCount, gSpecialRingCount);
+            return;
+        }
+    }
+}
+
+void Task_802EE78(void)
+{
+    StageResultsInit *sri = TASK_DATA(gCurTask);
+    struct TransitionState *ts = &sri->ts;
+    NextTransitionFrame(ts);
+
+    if (++sri->unk10 > 8) {
+        ts->unk4 = 0;
+        ts->unk2 = 2;
+        ts->speed = 0;
+        ts->unk8 = 191;
+        ts->unkA = 0;
+
+        if (gCurrentLevel != LEVEL_INDEX(ZONE_7, ACT_BOSS)) {
+            gFlags &= ~FLAGS_4;
+        }
+
+        if (IS_FINAL_STAGE(gCurrentLevel)) {
+            u32 x, y;
+
+            x = gUnknown_080D6DE4[sri->unk11][0];
+            x -= Q_24_8_TO_INT(gPlayer.x);
+            y = gUnknown_080D6DE4[sri->unk11][1];
+            y -= Q_24_8_TO_INT(gPlayer.y);
+
+            gPlayer.x += Q_24_8(x);
+            gPlayer.y += Q_24_8(y);
+
+            gCamera.x += x;
+            gCamera.y += y;
+            gCamera.unk20 += x;
+            gCamera.unk24 += y;
+            gCamera.unk10 += x;
+            gCamera.unk14 += y;
+
+            if (gUnknown_030056A4 != NULL) {
+                gUnknown_030056A4->posX += Q_24_8(x);
+                gUnknown_030056A4->posY += Q_24_8(y);
+            }
+
+            gUnknown_030055B0++;
+        }
+
+        gCurTask->main = Task_802F06C;
+    }
+}
+
+void sub_802EF68(s16 p0, s16 p1, u8 p2)
+{
+    struct Task *t = TaskCreate(Task_802EE78, sizeof(StageResultsInit), 0x6080, 0, NULL);
+    StageResultsInit *sri = TASK_DATA(t);
+    struct TransitionState *ts = &sri->ts;
+
+    sri->unk10 = 0;
+    sri->unkC = p0;
+    sri->unkE = p1;
+    sri->unk11 = p2;
+
+    ts->unk0 = 0;
+    ts->unk4 = 0x800;
+    ts->unk2 = 1;
+    ts->speed = 192;
+    ts->unk8 = 191;
+    ts->unkA = 0;
+
+    m4aSongNumStart(SE_333);
+}
+
+void InitHBlankBgOffsets(u16 xOffset)
```

Comment:
I wonder if these bg offset functions need to be named in a way which indicates they are related to results?

Reply:
Maybe...

Reply:
Sure, but are you sure that's not because it runs this function when it's about to show results?

Reply:
Or do you mean they've just used this function elsewhere as some sort of hack?